packages/@rnw-scripts/doxysaurus/src/config.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+//
+// Configuration for the Doxysaurus tool.
+//
+
+import path from 'path';
+import {promises as fs, existsSync} from 'fs';
+
+// Definition of a config.
+export interface Config {
+  configDir: string;
+  input: string;
+  output: string;
+  filePatterns?: string[];
+  prefix: string;
+  index?: string;
+  indexTemplate?: string;
+  projects?: string[];
+  types?: string[];
+  namespaceAliases?: [string, string[]][];
```

Comment:
Any reason why representing using nested arrays vs object literal? That shape is more common for map-like types in my experience.

E.g. `Record<string, string[]>`.

```js
 "namespaceAliases": {
    "winrt::Microsoft::ReactNative": ["React"]
  },
```

vs current 

```js
 "namespaceAliases": [
    ["winrt::Microsoft::ReactNative", ["React"]]
  ],
```

Reply:
Same for other maps

Reply:
I have started with object literal, but it has a an unfortunate issue that the 'constructor' key has a special meaning. As a result, it was giving me issues in some cases. I have replaced it with arrays to feed it to the Map class without any extra processing.

Reply:
I will see if can switch to Record style for Config instead.

Reply:
Changed to object literal in commit https://github.com/microsoft/react-native-windows/pull/7044/commits/464f21524bddd39a72e4037dd6a20de9dec2de62